3-enrich/stream-enrich/core/src/test/scala/com.snowplowanalytics.snowplow.enrich.stream/SpecHelpers.scala

```diff
@@ -141,123 +150,176 @@ object SpecHelpers {
-                |"mktSource": ["utm_source", "source"],
-                |"mktTerm": ["utm_term", "legacy_term"],
-                |"mktContent": ["utm_content"],
-                |"mktCampaign": ["utm_campaign", "cid", "legacy_campaign"]
-              |}
-            |}
-          |}
-        |},
-        |{
-          |"schema": "iglu:com.snowplowanalytics.snowplow/user_agent_utils_config/jsonschema/1-0-0",
-          |"data": {
-            |"vendor": "com.snowplowanalytics.snowplow",
-            |"name": "user_agent_utils_config",
-            |"enabled": true,
-            |"parameters": {
-            |}
-          |}
-        |},
-        |{
-          |"schema": "iglu:com.snowplowanalytics.snowplow/referer_parser/jsonschema/1-0-0",
-          |"data": {
-            |"vendor": "com.snowplowanalytics.snowplow",
-            |"name": "referer_parser",
-            |"enabled": true,
-            |"parameters": {
-              |"internalDomains": ["www.subdomain1.snowplowanalytics.com"]
-            |}
-          |}
-        |}
-      |]
-    |}""".stripMargin.replaceAll("[\n\r]","").stripMargin.replaceAll("[\n\r]","")
-
     val config = EnrichConfig(
       streams = StreamsConfig(
         InConfig("raw"),
-        OutConfig("enriched", "bad", "partitionkey"),
+        OutConfig("enriched", "pii", "bad", "partitionkey"),
         Kafka("brokers", 1),
         BufferConfig(1000L, 100L, 1200L),
         "appName"
       ),
-      monitoring = None)
-
-    val validatedResolver = for {
-      json <- JsonUtils.extractJson("", """{
-        "schema": "iglu:com.snowplowanalytics.iglu/resolver-config/jsonschema/1-0-0",
-        "data": {
+      monitoring = None
+    )
+    new TestSource(config, resolver, enrichmentRegistry, None)
+  }
+  val igluCentralDefaultConfig =
+    """{
+    "schema": "iglu:com.snowplowanalytics.iglu/resolver-config/jsonschema/1-0-0",
+    "data": {
 
-          "cacheSize": 500,
-          "repositories": [
-            {
-              "name": "Iglu Central",
-              "priority": 0,
-              "vendorPrefixes": [ "com.snowplowanalytics" ],
-              "connection": {
-                "http": {
-                  "uri": "http://iglucentral.com"
-                }
-              }
+      "cacheSize": 500,
+      "repositories": [
+        {
+          "name": "Iglu Central",
+          "priority": 0,
+          "vendorPrefixes": [ "com.snowplowanalytics" ],
+          "connection": {
+            "http": {
+              "uri": "http://iglucentral.com"
             }
-          ]
+          }
         }
-      }
-      """)
-      resolver <- Resolver.parse(json).leftMap(_.toString)
-    } yield resolver
+      ]
+    }
+  }
+  """
 
-    implicit val resolver: Resolver = validatedResolver.fold(
-      e => throw new RuntimeException(e),
-      s => s
-    )
+  val igluConfig = {
+    val resolverEnvVar = for {
+      config <- sys.env.get("ENRICH_RESOLVER_CONFIG")
```

Comment:
why is this needed? it should either be all env vars or no env vars?

Reply:
This how it is done in scala-common-enrich (it uses an env var to override the default iglu repo). I am not sure how it should be done otherwise (I don't understand all/no env vars).

Reply:
yes let's leave it as is ... as discussed it can be useful when testing and using a specific registry without having to modify the code itself